crates/db_schema/src/source/person.rs

```diff
@@ -45,18 +44,16 @@ pub struct Person {
   pub banner: Option<DbUrl>,
   /// Whether the person is deleted.
   pub deleted: bool,
-  #[cfg_attr(feature = "full", ts(skip))]
-  #[serde(skip, default = "placeholder_apub_url")]
-  pub inbox_url: DbUrl,
-  #[serde(skip)]
-  pub shared_inbox_url: Option<DbUrl>,
   /// A matrix id, usually given an @person:matrix.org
   pub matrix_user_id: Option<String>,
   /// Whether the person is a bot account.
   pub bot_account: bool,
   /// When their ban, if it exists, expires, if at all.
   pub ban_expires: Option<DateTime<Utc>>,
   pub instance_id: InstanceId,
+  #[cfg_attr(feature = "full", ts(skip))]
+  #[serde(skip, default)]
+  pub inbox_id: i32,
 }
```

Comment:
The best solution seems implementing the relevant traits manually instead of using derive, will give that a try.

Reply:
The shared inbox would move to the `site.inbox_url` then? You get that by joining from person.instance_id -> site.instance_id.

Reply:
The idea is to store only a single inbox per user, as we dont differentiate between normal inbox and shared inbox. And to avoid storing the same string many times, I moved it into a new table inbox. We cant use site.inbox_url directly, as other federated platforms may still use a separate inbox per user.

Anyway the main problem is that I cant join `person.inbox_id` with `inbox.id` in normal diesel queries, because the `Queryable` trait doesnt support joins.

Reply:
It sounds like a simple join should work fine, you'll just need to add a function to the person impls, that does the join. If you could show me where you need this joined data, or where you're calling it from, I could help.

Reply:
The query itself is not hard, problem is that `inbox_url` somehow needs to be part of `Person` which is not compatible with the derives. Otherwise I could select the inbox_url only when its actually needed, but [here](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/objects/person.rs#L208) it wont work because the function is not async. Maybe I need to change it to async in the library.